### PR TITLE
Donation form update

### LINKED ISSUE #2737: Project: refresh donate page (#proj-donor-page v1)
## Goal

Refresh the donate page in advance of the giving season, making it more appealing and steering people to the best provider.

[figma](https://www.figma.com/file/9bQW46okOjxeJtismQQdPu/%5BDesktop%5D-Screens-1.0?type=design&node-id=5301-1464&mode=design&t=yj5Nx8rJMkxKPt9a-0), [slide](https://docs.google.com/presentation/d/16AYWtqXkLGF-4iKyUj8VeYkROOdsKP2F3p6vpVswtFs/edit#slide=id.g288bed48aa0_0_36)

## Scope

**Necessary**
- [ ] #2744
- [ ] #2890

**Nice to have**
- [x] donate: capitalize names

### PR DESCRIPTION
**Pull request by mlbrgl**
closes #2737, #2890, #2744

- staging (see top of the stack)

This the [v1 of #proj-donor-page](https://owid.slack.com/archives/C060QED1YJV/p1698858247135849)

> we won't change the underlying infrastructure in this iteration
we won't change any of the flows or processes
adding names to /funding will still use the current semi-manual process
we won't change the thank-you page
we won't change the content or location (WP) of the /funding page
instead, we're going to focus on delivering the redesigned form and adding FAQs to /donate

- [x] check metadata (`showOnList`, `name`) is still surfaced and exported in `owid-donors`

### CODE REVIEW

**Comment by mlbrgl** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2888** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2888" target="_blank">Graphite</a> 
    * **PR #2889** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2889" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2889?utm_source=stack-comment).

File: site/stripe/DonateForm.tsx [hunk 1]
```diff
@@ -197,161 +202,137 @@ export class DonateForm extends React.Component {
...
-                                    id={code}
-                                    value={code}
-                                    name="currency"
-                                    onChange={() => this.setCurrency(code)}
-                                    checked={this.currencyCode === code}
-                                />
-                                <label htmlFor={code}>{code}</label>
-                            </div>
+                            <input
+                                type="button"
```
Comment (ikesau) [inline site/stripe/DonateForm.tsx:252]: These should start to wrap on mobile probably (this is 360w - apparently the width of a Galaxy S20)

image

File: site/stripe/DonateForm.tsx [hunk 2]
```diff
@@ -75,24 +76,22 @@ export class DonateForm extends React.Component {
...
         this.customAmount = amount
-        this.isCustom = true
-    }
-
-    @action.bound setIsCustom(isCustom: boolean) {
-        this.isCustom = isCustom
+        this.presetAmount = undefined
     }
 
     @action.bound setName(name: string) {
```
Comment (ikesau) [inline site/stripe/DonateForm.tsx:87]: [This is an enormous can of worms](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/), but I think we should trim this input. Right now, `" "` is a valid name.

File: site/stripe/DonateForm.tsx [hunk 3]
```diff
@@ -366,26 +347,114 @@ export class DonateForm extends React.Component {
...
-                    disabled={this.isLoading}
-                >
-                    Donate{" "}
-                    {this.amount ? `${this.currencySymbol}${this.amount}` : ""}{" "}
-                    {this.interval === "monthly" ? "per month" : ""}
-                </button>
-
-                <p className="note">
+                <div className="donation-payment">
+                    <button
```
Comment (ikesau) [inline site/stripe/DonateForm.tsx:351]: Just confirming [we're okay with the `owid-donate` Netlify function](https://github.com/owid/owid-grapher/issues/2751) for the time being? 

Should we make a ticket to migrate that, if Donation form v3 is a while away?

File: site/stripe/DonateForm.tsx [hunk 4]
```diff
@@ -197,161 +202,137 @@ export class DonateForm extends React.Component {
...
                         </div>
                     </div>
                 </fieldset>
 
-                <fieldset>
-                    <div className="owid-block-field">
-                        <label htmlFor="name">
-                            <h3>Your name (optional)</h3>
+                <fieldset className="donation-public-checkbox">
+                    <Checkbox
```
Comment (ikesau) [inline site/stripe/DonateForm.tsx:303]: This is very nitpicky, but the rest of the form is _so nicely done_ that I feel compelled to mention that this is the only form element that lacks a focus indicator when navigating by keyboard.

**Review by ikesau — CHANGES_REQUESTED** [review]
Very nice! 🙂 

Tested with FF, Chrome and Safari, and only found that Currency wrapping issue.

Otherwise, there are a few places where a little bit of validation might help some people. Happy to discuss further 🙂

File: site/stripe/DonateForm.tsx (hunk 3 above)
Comment (mlbrgl) [reply site/stripe/DonateForm.tsx:351]: yes that's still the plan. I created two issues to track this:
- #2908
- #2909

File: site/stripe/DonateForm.tsx (hunk 1 above)
Comment (mlbrgl) [reply site/stripe/DonateForm.tsx:252]: 👍 Moving to a 2 col grid on small screens.

**Comment by mlbrgl** [comment]
Thanks for the speedy review @ikesau!

All of your comments make sense to me, and I've captured their essence in new issues listed in #2908.

I won't act on them right away though, as all inputs (text fields, checkbox) will disappear in #2908, but I made an explicit note in this issue that we should revisit those comments if v2 gets delayed/deprioritized.

**Comment by mlbrgl** [comment]
### Merge activity

* **Nov 9, 3:32 AM**: @mlbrgl started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2889).

**Pull request merged**